Bus Scheduler/src/components/tripSearch/store/hook.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { GET } from '../../../store/api/axios'
+import { tripListApiUrl } from '../../../store/api/urls'
+
+import { useCustomContext } from '../../../store/contextProvider';
+import { TripSearchContext } from './tripSearchProvider';
+
+export const useFetchTrips = () => {
+    const { data, actions } = useCustomContext(TripSearchContext);
+
+    useEffect(() => {
+        const fetchTrips = async () => {
+            const response = await GET(tripListApiUrl);
+            return response;
+        }
+        const tripList = fetchTrips();
+
+        tripList.then((response) => {
+            actions.listTrips({ data: response });
+        })
```

Comment:
good to follow delcarative code, encapsulate the code in a function and use it in useEffect

Reply:
Done